src/pages/projectSettings/tabs/NotificationsTab/types.ts

```diff
@@ -6,30 +7,30 @@ export interface FormState {
   };
   subscriptions: Array<{
     subscriptionData: {
-      event: {
-        eventSelect: string;
-        extraFields: { [key: string]: string };
-        regexSelector: { [key: string]: string }[];
+      event?: {
+        eventSelect?: string;
+        extraFields?: { [key: string]: string };
+        regexSelector?: FormRegexSelector[];
       };
-      notification: {
-        notificationSelect: string;
-        jiraCommentInput: string;
-        slackInput: string;
-        emailInput: string;
-        jiraIssueInput: {
-          projectInput: string;
-          issueInput: string;
+      notification?: {
+        notificationSelect?: string;
+        jiraCommentInput?: string;
+        slackInput?: string;
+        emailInput?: string;
+        jiraIssueInput?: {
+          projectInput?: string;
+          issueInput?: string;
         };
-        webhookInput: {
-          urlInput: string;
-          secretInput: string;
```

Comment:
Set all of these props to be optional to reflect RJSF default behavior of only setting form values once they are used. 
This wasn't needed elsewhere because we declare a default state for the form so the values are always set. For this tab since we create the forms on the fly we don't have a default state for any field.
https://github.com/khelif96/spruce/blob/90692b792e45d65754ff9ed773b78c0df8ae2ca5/src/components/Notifications/index.tsx#L74-L86